secboot/encrypt_tpm.go

```diff
@@ -21,14 +21,38 @@
 package secboot
 
 import (
+	"crypto/rand"
+	"os"
+	"path/filepath"
+
 	sb "github.com/snapcore/secboot"
+
+	"github.com/snapcore/snapd/osutil"
 )
 
 var (
 	sbInitializeLUKS2Container       = sb.InitializeLUKS2Container
 	sbAddRecoveryKeyToLUKS2Container = sb.AddRecoveryKeyToLUKS2Container
 )
 
+type RecoveryKey sb.RecoveryKey
```

Comment:
why type alias this at all? why not just use sb.RecoveryKey everywhere?

Reply:
That was done to keep usage similar to EncryptionKey. The snapcore/secboot RecoveryKey type doesn't have an initialization method and having NewRecoveryKey() returning an external type seems strange. Also this allows us to add our own methods (currently only Save() but this could be implemented as a regular function).

Reply:
Feels ok, especially since we are adding new methods on top. I'm only bit worried that the type will not be define when we build with `nosecboot` tag, what may trigger some confusing errors.

Reply:
The places we use this type should be entirely on `!nosecboot` territory (under `gadget/install` and in secboot `*_tpm` files).